torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -219,19 +217,19 @@ def precompile(self, warm_cache_only_with_cc=None):
 
             seen_configs = set(self.configs)
 
-            device_prop = self.device_interface.Worker.get_device_properties(
-                self.triton_meta["device"]
-            )
+            device_prop = self.device_props
             if (
                 self.inductor_meta.get("dynamic_scale_rblock", True)
                 and self.heuristic_type == HeuristicType.REDUCTION
                 and self.size_hints is not None
-                # Disable for AMDGPU as Triton is not ready to return n_regs for a compiled_binary.
-                and not self.inductor_meta.get("is_hip")
-                # Disable for Intel GPU as Triton is not ready to return n_regs for a compiled_binary.
-                and self.device_type != "xpu"
+                # Disable for AMDGPU/Intel as Triton is not ready to return n_regs for a compiled_binary.
+                and device_prop.type == "cuda"
```

Comment:
Just checking this is what you want. Instead of != xpu and != hip, you're doing == cuda. I assume that's fine?

Reply:
Yes, these are the only 3 supported values and if we add new ones we likely want to skip this.